reader.go

```diff
@@ -40,14 +40,22 @@ func (s *shellReader) rlPrompt() string {
 	return ""
 }
 
-func (s *shellReader) readPassword() string {
+func (s *shellReader) readPasswordErr() (string, error) {
 	prompt := ""
 	if s.buf.Len() > 0 {
 		prompt = s.buf.String()
 		s.buf.Truncate(0)
 	}
-	password, _ := s.scanner.ReadPassword(prompt)
-	return string(password)
+	password, err := s.scanner.ReadPassword(prompt)
+	if err != nil {
```

Comment:
minor but this error check is not needed. You can as well `return string(password), err)`.